lib/spack/spack/util/path.py

```diff
@@ -103,36 +103,27 @@ def _get_padding_string(length):
     return os.path.sep.join(reps_list)
 
 
-def _add_computed_padding(path):
-    """Subtitute in padding of os-specific length.  The intent is to leave
-    SPACK_MAX_INSTALL_PATH_LENGTH characters available for parts of the
-    path generated by spack.  This is to allow for not-completely-known
-    lengths of things like os/arch, compiler, package name, hash length,
-    etc.
-    """
-    padding_regex = re.compile(r'(\$[\w\d\:]+\b|\$\{[\w\d\:]+\})')
-    m = padding_regex.search(path)
-    if m and m.group(0).strip('${}').startswith('padding'):
-        padding_part = m.group(0)
-        len_pad_part = len(m.group(0))
-        p_match = re.search(r'\:(\d+)', padding_part)
-        if p_match:
-            computed_padding = _get_padding_string(int(p_match.group(1)))
-        else:
-            # Take whatever has been computed/substituted so far and add some
-            # room
-            path_len = len(path) - len_pad_part + SPACK_MAX_INSTALL_PATH_LENGTH
-            system_max_path = get_system_path_max()
-            needed_pad_len = system_max_path - path_len
-            computed_padding = _get_padding_string(needed_pad_len)
-        return padding_regex.sub(computed_padding, path)
-    return path
+def add_padding(path, length):
+    """Add padding subdirectories to path until total is length characters
+
+    Returns the padded path. If path is length - 1 or more characters long,
+    returns path. If path is length - 1 characters, warns that it is not
+    padding to length
+
+    Assumes path does not have a trailing path separator"""
+    padding_length = length - len(path) - 1  # -1 for separator
```

Comment:
(question) This logic handling the possibility of the padding length being only 1 larger seems new. Should it raise an exception? I assume if someone asks for padding, they need it.

Reply:
The reason it's new is because previously, the padding didn't have to start a new directory name, so we could patch `foo/` to `foos/`. Now that we're storing the sbang at the top level in `foo/bin/sbang`, there is no way to pad `foo` to 4 characters. It's just not possible.

I'd be fine with having it raise an exception, but my guess is that it will A) never come up, and B) not be useful to users. I don't know of any use case for padding that actually depend on the exact number.